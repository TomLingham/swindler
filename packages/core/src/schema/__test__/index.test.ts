import { request, response } from "..";

describe("mock schema", () => {
  describe("requestSchema", () => {
    test("normalization", async () => {
      const value = { path: "/" };
      if (request.isValid(value)) {
        const result = await request.normalize(value);
        expect(result).toMatchSnapshot();
      } else {
        fail(await request.validate(value));
      }
    });

    test("'path' is required", () => {
      return expect(request.validate({})).resolves.toMatchSnapshot();
    });

    test("'method' defaults to 'GET'", async () => {
      const value = { path: "/" };
      if (request.isValid(value)) {
        const result = await request.normalize(value);
        expect(result.method).toBe("GET");
      } else {
        fail();
      }
    });
  });

  describe("responseSchema", () => {
    test("normalization", async () => {
      const value = {};
      if (response.isValid(value)) {
        const result = await response.normalize(value);
        expect(result).toMatchSnapshot();
      } else {
        fail(await response.validate(value));
      }
    });

    test("'status' defaults to 200", async () => {
      const value = { headers: {} };
      if (response.isValid(value)) {
        const result = await response.normalize(value);
        expect(result.status).toBe(200);
      } else {
        fail(await response.validate(value));
      }
    });
  });
});
