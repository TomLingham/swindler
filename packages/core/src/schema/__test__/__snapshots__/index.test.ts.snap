// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock schema requestSchema 'path' is required 1`] = `
Array [
  "path is a required field",
]
`;

exports[`mock schema requestSchema normalization 1`] = `
Object {
  "method": "GET",
  "path": "/",
}
`;

exports[`mock schema responseSchema normalization 1`] = `
Object {
  "headers": Object {},
  "status": 200,
}
`;
